src/NumericalAlgorithms/SphericalHarmonics/StrahlkorperFunctions.cpp

```diff
@@ -415,6 +416,25 @@ void normal_one_form(const gsl::not_null<tnsr::i<DataVector, 3, Fr>*> one_form,
   }
 }
 
+template <typename Fr>
+std::vector<std::array<double, 4>> fit_ylm_coeffs(
+    const DataVector& times,
+    const std::vector<Strahlkorper<Fr>>& strahlkorpers) {
+  size_t num_observations = strahlkorpers.size();
+  size_t num_coefficients =
+      2.0 * (strahlkorpers[0].l_max() + 1.0) * (strahlkorpers[0].m_max() + 1.0);
+  intrp::LinearLeastSquares<3> predictor(num_observations);
+  std::vector<DataVector> coefficients;
+  for (size_t i = 0; i < num_coefficients; ++i) {
+    DataVector same_coeff_of_each_strahlkorper(num_observations);
+    for (size_t j = 0; j < num_observations; ++j) {
+      // Contains the same coefficient of each strahlkorper in the time series.
```

Comment:
need to check that all strahlkorpers have the same l_max and m_max or there is going to be big trouble...